test/meson.build

```diff
@@ -0,0 +1,49 @@
+gtest_dep = dependency('gtest', main: true, disabler: true, required: false)
+gmock_dep = dependency('gmock', disabler: true, required: false)
+if not gtest_dep.found() or not gmock_dep.found()
+    cmake = import('cmake')
+    gtest_opts = cmake.subproject_options()
+    gtest_opts.set_override_option('warning_level', '1')
+    gtest_opts.set_override_option('werror', 'false')
+    gtest_proj = cmake.subproject('googletest',
+                                  options: gtest_opts,
+                                  required: false)
+    if gtest_proj.found()
+        gtest_dep = declare_dependency(
+            dependencies: [
+                dependency('threads'),
+                gtest_proj.dependency('gtest'),
+                gtest_proj.dependency('gtest_main'),
+            ]
+        )
+        gmock_dep = gtest_proj.dependency('gmock')
+  else
+        assert(
+            not get_option('tests').allowed(),
+            'Googletest is required if tests are enabled'
+        )
+  endif
+endif
+
+dependecy_list = [gtest_dep, gmock_dep, sdbusplus]
```

Comment:
1. spellcheck: dependency*
2. I don't think we need sdbusplus dependency for test cases. Can you remove it and see if you get any compile time errors

Reply:
1. Updated the spellcheck 
2. sdbusplus dependency coming from utils.cpp file